### ISSUE TITLE
Questions about the frequent segmentation

### ISSUE THREAD

**Issue by ORainn** [juanmc2005/StreamingSpeakerDiarization#11]
Hi there, I'm using your codes to process a wav file. In the results, I found that when the same speaker speaks continuously, his words will be divided into several paragraphs, like the following:

`SPEAKER output 1 4.953 1.249 <NA> <NA> speaker0 <NA> <NA>
SPEAKER output 1 6.455 1.342 <NA> <NA> speaker0 <NA> <NA>
SPEAKER output 1 8.691 0.253 <NA> <NA> speaker0 <NA> <NA>`

Such frequent segmentation makes my subsequent speech recognition more difficult. So my question is, is it possible to speak without segmentation when the same speaker is speaking? Or is there any other way to achieve this effect?

Thank you.

**Comment by juanmc2005** [comment]
Hello! Since a speaker can make pauses while speaking, it is normal that silences appear between two speech segments. The duration of these pauses may however not be adapted to your downstream task, but in an end-to-end segmentation model this is ultimately defined by the annotations of the training data.

You could try to fine-tune the segmentation model with in-domain data and appropriate labels, but that may be too costly.
If that's not an option, you can try to merge segments using `Annotation.support(collar)`, where `collar` is the maximum duration that you allow for a pause within the same speaker.

In the demo, you can play with this collar in the [OutputBuilder](https://github.com/juanmc2005/StreamingSpeakerDiarization/blob/main/src/sinks.py#L18), notice how it's used in [this line](https://github.com/juanmc2005/StreamingSpeakerDiarization/blob/main/src/sinks.py#L104), in case you're implementing your own sink.

Let me know if this answers your question.

**Comment by ORainn** [comment]
Hello! I just tried your suggestion on using `Annotation.support(collar)`. I modified the value of 'collar' to a larger value to get more continuous segmentation fragments, which can effectively solve the problem I raised and better guarantee the downstream work.

Thank you very much for helping me so effectively! I wish you all the best!
